pkg/asset/installconfig/azure/dns.go

```diff
@@ -173,3 +191,23 @@ func (client *ZonesClient) GetAllPublicZones() (map[string][]string, error) {
 	}
 	return allZones, nil
 }
+
+func (client *RecordSetsClient) GetRecordSet(publicZoneID string, relativeRecordSetName string, recordType azdns.RecordType) (*azdns.RecordSet, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer cancel()
+
+	parts := strings.Split(publicZoneID, "/")
+	if len(parts) != 9 {
+		return nil, errors.New("invalid zone ID")
+	}
+
+	rgName := parts[4]
+	zoneName := parts[8]
+
+	recordset, err := client.azureClient.Get(ctx, rgName, zoneName, relativeRecordSetName, recordType)
```

Comment:
I might be missing it, but is this taking into consideration the region? And the fact that the zone could exist in a RG different than the rest of the deployment?

Reply:
Yes. This was supposed to happen but slipped. Fixed!

Reply:
I am not sure I follow in regards to the region, can you elaborate? Zones are unique per domains + resource group, so this will pull recordsets from a specify zone + resource group combo. If there is an identical zone in another resource group, it will not be the same.

Reply:
Ok, I made it clear already, just checked that the `rgName` here is the one in  `BaseDomainResourceGroupName`, so you have the right resource group.